app/models/bookmark.rb

```diff
@@ -8,12 +8,71 @@ class Bookmark < ActiveRecord::Base
     "reminder_type" # TODO 2021-04-01: remove
   ]
 
+  # Should only be created via the Bookmark.register_bookmarkable
+  # method; this is used to let the BookmarkQuery class query and
+  # search additional bookmarks for the user bookmark list, and
+  # also to enumerate on the registered Bookmarkable types.
+  #
+  # Anything other than types registered in this way will throw an error
+  # when trying to save the Bookmark record  (excluding Post and Topic
+  # bookmarkables, which have special logic handled directly in core.)
+  class Bookmarkable
```

Comment:
Instead of interleaving  classes within the same file, I think it'll be better to move this to `concerns`.

Reply:
@tgxworld Is it really a concern though :thinking: ? It won't extend ActiveSupport::Concern, it's just a normal class. I'm not sure where to put it but I agree it shouldn't be in here.

Reply:
You're right, concern is probably not the right place. How about `app/services`? The folder is just meant to hold plain old ruby objects that are aware of business logic.

Reply:
Cool will move it there :)